### PR TITLE
add julia coding guidelines

### PR DESCRIPTION
**Pull request by MaHaWo**
- used the Python guidelines as a template and filled int the equivalent Julia stuff. 
- added a section with links to official coding guidelines. Julia is based on a little bit different paradigms, so thought it might be useful to have those if people come over from, say, Python.

### CODE REVIEW

**Comment by MaHaWo** [comment]
@lkeegan @iulusoy if you have time, maybe you could have a look if it's ok like this?

File: julia/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+- Machine Learning/Deep Learning: 
+    - [SciKitLearn.jl](https://github.com/cstjean/ScikitLearn.jl) for having access to the python based [scikit-learn](https://scikit-learn.org/) library in Julia. 
+    - [Flux.jl](https://github.com/FluxML/Flux.jl) for deep learning applications. 
+
+Have a look at [the JuliaHub package database](https://juliahub.com/ui/Packages) to find other packages for your use case. 
+
+
+## Style guides and best-practices:
+
+Much has already been written about how to write good Julia code which needs not be repeated here. Here are some starging points:
```
Comment (lkeegan) [inline julia/README.md]:
```suggestion
Much has already been written about how to write good Julia code which needs not be repeated here. Here are some starting points: 
```

File: julia/README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    - [Plots.jl](https://github.com/JuliaPlots/Plots.jl): Julia Frontend with a homogeneous interface for various well known plotting packages like [matplotlib](https://matplotlib.org/), [plotly](https://plotly.com/) or [GR](https://gr-framework.org/#). 
+    - [Makie.jl](https://github.com/MakieOrg/Makie.jl): Feature-complete plotting package in pure Julia. 
+    - [AlgebraOfGraphics.jl](https://github.com/MakieOrg/AlgebraOfGraphics.jl) Provides support for the grammar-of-graphics paradigm based on Makie.jl .
+    - [VegaLite.jl](https://github.com/queryverse/VegaLite.jl): Provides access to [VegaLite](https://vega.github.io/vega-lite/) from Julia, particularly useful for making interactive graphics. 
+
+- Machine Learning/Deep Learning: 
+    - [SciKitLearn.jl](https://github.com/cstjean/ScikitLearn.jl) for having access to the python based [scikit-learn](https://scikit-learn.org/) library in Julia. 
+    - [Flux.jl](https://github.com/FluxML/Flux.jl) for deep learning applications. 
+
+Have a look at [the JuliaHub package database](https://juliahub.com/ui/Packages) to find other packages for your use case.
```
Comment (lkeegan) [inline julia/README.md]: The link is correct if I put the address in a browser, but if I click on this link I get this error:

image

Just a note to check this link works after merging the changes

**Review by lkeegan — COMMENTED** [review]
thanks, lgtm!

File: julia/README.md (hunk 2 above)
Comment (MaHaWo) [reply julia/README.md]: hm, that's strange. tried it from code editor and it worked. Also tried to replace the url, no luck.

**Pull request merged**